src/Radiation/rrtmgp_interface.jl

```diff
@@ -0,0 +1,188 @@
+    params = RRTMGPParameters(eltype(grid))
+    # TODO!: need helpers here for reshaping fields to column layout
+    sfc_emis = reshape(sfc_emission, Nbnd, Ncols)
+    lw_inc_flux = lw_inc_flux isa Nothing ? nothing : reshape(lw_inc_flux, Nbnd, Ncols)
+
+    return TwoStreamLWRTE(grid_params; params, sfc_emis, inc_flux=lw_inc_flux)
+end
+
+"""
+    RTE.TwoStreamSWRTE(grid; 
+                       cos_zenith, 
+                       toa_flux, 
+                       sfc_alb_direct,
+                       inc_flux_diffuse, 
+                       sfc_alb_diffuse, 
+                       inc_flux_diffuse, 
+                       isothermal_boundary_layer)
+
+Construct the two-stream shortwave RTE. Arrays are reshaped to column layout:
+- `cos_zenith`, `toa_flux`: (Nx, Ny) → (Ncols)
+- `sfc_alb_direct`, `sfc_alb_diffuse`: (Nbnd, Nx, Ny) → (Nbnd, Ncols)
+- `inc_flux_diffuse`: nothing or (Nbnd, Nx, Ny) → (Nbnd, Ncols)
+- `isothermal_boundary_layer`: Bool
+"""
+function RRTMGP.RTE.TwoStreamSWRTE(
+    grid::RectilinearGrid;
+    cos_zenith,
+    toa_flux,
+    sfc_alb_direct,
+    sfc_alb_diffuse,
+    inc_flux_diffuse,
+    isothermal_boundary_layer
+)
+    Ncols = ncols(grid)
+    Nbnd = size(sfc_alb_direct, 1)
+    grid_params = RRTMGPGridParams(grid; isothermal_boundary_layer)
+
+    # TODO!: need helpers here for reshaping fields to column layout
+    cosz = reshape(cos_zenith, Ncols)
+    toa = reshape(toa_flux, Ncols)
+    alb_dir = reshape(sfc_alb_direct, Nbnd, Ncols)
+    alb_diff = reshape(sfc_alb_diffuse, Nbnd, Ncols)
+    inc_flux_diffuse = inc_flux_diffuse isa Nothing ? nothing : reshape(inc_flux_diffuse, Nbnd, Ncols)
+
+    return TwoStreamSWRTE(
+        grid_params; 
+        cos_zenith=cosz, 
+        toa_flux=toa, 
+        sfc_alb_direct=alb_dir, 
+        inc_flux_diffuse=inc_flux_diffuse,
+        sfc_alb_diffuse=alb_diff,
+    )
+end
+
+"""
+    update_atmospheric_state!(as::GrayAtmosphericState, temperature::Field, pressure::Field)
+
+Update the atmospheric state with new temperature and pressure fields.
+"""
+function update_atmospheric_state!(as::RRTMGP.AtmosphericStates.GrayAtmosphericState, temperature::Field, pressure::Field)
+    grid = temperature.grid
+    Nz = size(grid, 3)
+    Ncols = ncols(grid)
+
+    # TODO!: need helpers here for reshaping fields to column layout
+    as.p_lay .= reshape(PermutedDimsArray(interior(pressure), (3, 1, 2)), Nz, Ncols)
+    as.p_lev .= reshape(PermutedDimsArray(interior(Field(@at (Center, Center, Face) pressure)), (3, 1, 2)), Nz + 1, Ncols)
+    as.t_lay .= reshape(PermutedDimsArray(interior(temperature), (3, 1, 2)), Nz, Ncols)
+    as.t_lev .= reshape(PermutedDimsArray(interior(Field(@at (Center, Center, Face) temperature)), (3, 1, 2)), Nz + 1, Ncols)
+    as.t_sfc .= view(as.t_lev, 1, :)
+
+    return nothing
+end
+
+"""
+    context_from_arch(arch)
+
+Return a `ClimaComms` execution context compatible with the provided
```

Comment:
what's an "execution context"?

Reply:
@bischtob do you understand the distinction between "context" and "architecture"?

Reply:
I believe they are equivalent, but I am not 100% certain in all cases (e.g., distributed).